### DISCUSSION TITLE
What is this blue grid and how to disable it?

### DISCUSSION THREAD

**Discussion by inexcode** [TokisanGames/Terrain3D#493] [Questions] [answered]
See the screenshot. I think I didn't enable anything, and I've checked every option I see. This grid also shows up in the game view.

![Screenshot from 2024-09-16 20-33-55](https://github.com/user-attachments/assets/8703123c-5901-46f6-8b0e-9e8f09cdc9a1)

**Comment by TokisanGames** [comment]
I've never seen that before in Godot or Terrain3D. Are you 100% sure it is caused by Terrain3D? What versions of Godot and Terrain3D? GPU and OS? GPU Driver version? Renderer? Do you see this in the demo?

**Reply by inexcode** [reply]
I'm not sure at this point...
It happened on my Framework 13 laptop (11th gen Intel, Iris integrated graphics, NixOS native installation, Godot 4.3, Terrain3D 0.9.2-beta) with this random jpg grass texture. At first, it doesn't show up, but after I reload the project, these lines appear. Their transparency depends on the distance from the camera and smoothly animated.
But when I switched the texture to something from PolyHaven and packed textures with Terrain3D's tool, these lines no longer appear with the new texture.

Then, I tried to replicate this on my primary rig (NixOS, Ryzen 5950X, AMD Radeon RX 7900 XT, Godot 4.3 installation from Steam, Terrain3D 0.9.2-beta) and it is all the same.

It looks like lines appear on the texture borders, as their density changes when I change the UV Scale property.

Renderer is Forward+ in all cases.

Here I'm attaching the texture that causes the issue:
![grass-2](https://github.com/user-attachments/assets/8f291dc5-c42b-4b0c-8ac3-2993afc349c8)

**Reply by TokisanGames** [reply, marked as answer]
You didn't answer me about the demo, so I assume it works well.

I imported your grass into the demo for both albedo textures, generated mipmaps, and it works fine in Godot 4.2 and 4.3.

![{861782EB-597B-47D4-9450-BB306E9BF0E7}](https://github.com/user-attachments/assets/fb7e8c5c-8710-491e-8b66-71943b7cd368)

The software and this texture file work fine. The issue is either related to your texture prep process, or bad hardware, or old drivers.

> Their transparency depends on the distance from the camera and smoothly animated.
> But when I switched the texture to something from PolyHaven and packed textures with Terrain3D's tool, these lines no longer appear with the new texture.
> It looks like lines appear on the texture borders, as their density changes when I change the UV Scale property.

That's a problem with your texture preparation. We used to have strict requirements but now allow you stick any garbage textures in there as long as they match size and format. The responsibility for getting good results falls to you. To do so, you need quality textures, then must prepare them properly, then set the correct import settings. We have all of this documented. I see no mention in your post about normals or roughness maps, and 800 is a weird size. Still, it works fine for me when imported properly. Please review the documentation and recommended practices and use quality textures.

**Reply by inexcode** [reply]
Looks like it is a problem with texture preparation indeed. Thank you for your help!
